File: DSA/50_Days_50_Questions/169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        n = len(nums)
        ans = nums[0]
        frq = 0 

        for i in range(n):
            if(frq == 0):
                ans = nums[i]
            if(ans == nums[i]):
                frq+=1
            else:
                frq-=1
        return ans

File: DSA/50_Days_50_Questions/169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int ans = nums[0];
        int frq = 0;

        for(int i=0; i<n; i++){
            if(frq == 0){
                ans = nums[i];
            }
            if(ans == nums[i]){
                frq++;
            }else{
                frq--;
            }
        }
        return ans ;
    }
};